src/qrad3/trace.c

```diff
@@ -85,7 +85,7 @@ void MakeTnodes (dmodel_t *bm)
 {
 	// 32 byte align the structs
 	tnodes = malloc( (numnodes+1) * sizeof(tnode_t));
-	tnodes = (tnode_t *)(((int)tnodes + 31)&~31);
+  memset(tnodes, 1, sizeof tnodes);
```

Comment:
Indention is wrong. Otherwise, I'm not 100% sure either if that code is really equivalent. What an ugly hack...

Reply:
It definitely isn't. The original changes the pointer to align it, the new one sets the contents of the memory.

Reply:
BTW the proper way to do this is either `posix_memalign` or `aligned_alloc`. Annoyingly this is not entirely standard, at least until newer C/C++ versions.

Reply:
Ah that actually makes a lot more sense than whatever I was thinking - I've attempted another fix and fixed up code the alignment, I'll test it on my PC at some point and see if it holds up :)